op-preimage/oracle.go

```diff
@@ -75,28 +72,30 @@ func (o *OracleServer) NextPreimageRequest(getPreimage PreimageGetter) error {
 		return fmt.Errorf("failed to serve pre-image %s request: %w", hex.EncodeToString(key[:]), err)
 	}
 
-	//add preimage k,v to the Preimage map for dumping it to json file
-	bytes := make([]byte, 32)
-	copy(bytes[:], key[:])
-	Preimages[hex.EncodeToString(bytes)] = hex.EncodeToString(value)
-
-	//write length & data to preimages.bin
-	binary.Write(PreimageFile, binary.BigEndian, uint64(len(value)))
-	for i := 0; i < len(value); i++ {
-		var ii = value[i]
-		err := binary.Write(PreimageFile, binary.LittleEndian, ii)
-		if err != nil {
-			return fmt.Errorf("failed to dump pre-image binary file: %w", err)
-		}
-	}
-	//padding some zeros to make preimages length can be divided by 8
-	if len(value)%8 != 0 {
-		for i := 0; i < 8-len(value)%8; i++ {
-			err := binary.Write(PreimageFile, binary.LittleEndian, byte(0))
+	if PreimageFile != nil {
+		//add preimage k,v to the Preimage map for dumping it to json file
+		bytes := make([]byte, 32)
+		copy(bytes[:], key[:])
+		Preimages[hex.EncodeToString(bytes)] = hex.EncodeToString(value)
+
+		//write length & data to preimages.bin
+		binary.Write(PreimageFile, binary.BigEndian, uint64(len(value)))
+		for i := 0; i < len(value); i++ {
+			var ii = value[i]
+			err := binary.Write(PreimageFile, binary.LittleEndian, ii)
 			if err != nil {
 				return fmt.Errorf("failed to dump pre-image binary file: %w", err)
 			}
 		}
+		//padding some zeros to make preimages length can be divided by 8
+		if len(value)%8 != 0 {
+			for i := 0; i < 8-len(value)%8; i++ {
```

Comment:
The code can be simplified to `PreimageFile.write(make([]byte, 8-len(value)%8))`

Reply:
Fixed in https://github.com/ethstorage/optimism/pull/9/commits/2a0a9fae2fde90bb7107456afd0b62a8bbbf8db0